src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,95 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React from 'react';
+import { TodoStatus } from '../../types/TodoStatus ';
+
+type Props = {
+  onChangeQuery: (query: string) => void,
+  onChangeSelect: (event: TodoStatus) => void,
+  query: string,
+  selectedOption: TodoStatus,
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  onChangeQuery = () => {},
+  onChangeSelect = () => {},
+  query,
+  selectedOption,
+}) => {
+  const onChangeSetQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onChangeQuery(event.target.value);
+  };
+
+  const removeQuery = () => onChangeQuery('');
+
+  const handleChangeOption
+  = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    let valueEnum: TodoStatus;
+
+    if (event.target.value === 'active') {
+      valueEnum = TodoStatus.Active;
+    } else if (event.target.value === 'completed') {
+      valueEnum = TodoStatus.Completed;
+    } else {
+      valueEnum = TodoStatus.All;
+    }
+
+    onChangeSelect(valueEnum);
+  };
+
+  const todoStatusValues = Object.values(TodoStatus);
+  const todoStatuses = Object.keys(TodoStatus)
+    .map((todoKey, i) => ([todoKey, todoStatusValues[i]]));
```

Comment:
If I understand correctly what you want to do, it will be enough for you to use Object.entries once:
```suggestion
  const todoStatuses = Object.entries(TodoStatus);
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries